File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
import collections
from collections import defaultdict
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pre=0
        cnt=0
        d=defaultdict(int)
        d[0]=1
        for i in range(len(nums)):
            pre+=nums[i]
            rem=pre-k
            cnt+=d[rem]
            d[pre]+=1
        return cnt

        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int n=nums.length;
        HashMap<Integer,Integer> prefixsumcount=new HashMap<>();
        int count=0;
        int prefixsum=0;
                prefixsumcount.put(0, 1);   
        for(int i=0;i<n;i++){
         prefixsum+=nums[i];
         int rem=prefixsum-k;
         if(prefixsumcount.containsKey(rem)){
            count+=prefixsumcount.get(rem);
         }   
  prefixsumcount.put(prefixsum, 
                prefixsumcount.getOrDefault(prefixsum, 0) + 1);        }
        return count;
    }
}